orchestrator/relayer/src/batch_relaying.rs

```diff
@@ -107,10 +107,10 @@ async fn get_batches_and_signatures(
             );
         }
     }
-    // reverse the list so that it is oldest first, we want to submit
-    // older batches so that we don't invalidate newer batches
+    // sort the batches by nonce in ascending order, we want to submit
+    // older batches first so that we don't invalidate newer batches
     possible_batches.par_iter_mut().for_each(|(_key, value)| {
-        value.reverse();
+        value.sort_by(|a, b| a.batch.nonce.cmp(&b.batch.nonce));
```

Comment:
Does this consideration need to be applied to valsets or logic calls in the relayer?

Reply:
Logic calls already handled sorting, or tried to, but I corrected it from a naive reversal to actual sorting by nonce (see `get_latest_logic_calls`).

For valsets, we specifically get the "latest" valset from the query `latest_signer_set_tx()` so no ordering is needed.

Reply:
Because signer set TXs are created based on checking power differential, there can be multiple pending TXs. If the relayer is only considering the latest one, I imagine that could lead to a bridge halt. AFAIK there is no mechanism cancelling older signer sets when newer ones go through, unless I'm missing something.

Reply:
The relayer uses `get_latest_valset` for the first pass, then from there decrements the nonce and uses `get_valset` until it finds an appropriate candidate.

Reply:
Ah. Just a bit weird that the sorting of logic calls is inside `get_latest_logic_calls` rather than the logic call relaying, but we're applying the sorting for batches inside the relayer but not the source get function. Should other callers of `get_latest_transaction_batches` have the same ordering applied to it?

Reply:
It looks like only the relayer is using it, so putting the sorting logic in one place or the other would just be for consistency.

Reply:
Ah, yes it is inconsistent. I'll move the batch sorrting inside of `get_latest_transaction_batches`

Reply:
Also removes the need for the parallelized sort of the hashmap values.